### PR TITLE
element x y integration, extracted from Note (v5).

### PR DESCRIPTION
**Pull request by rvilarl**
No visual diferencies, once merged I will rebase #34 and the number of modified files will decrease.

### CODE REVIEW

**Comment by ronyeh** [comment]
Thanks for these. I have been looking at the other PRs but as you are aware, the bigger PRs are more difficult for a reviewer to tackle. :-) Smaller bites are easier to understand.

Apologies if the reviews are going a bit slowly. "You're coding too fast!" Hahahah.

File: src/element.ts [hunk 1]
```diff
@@ -521,7 +544,11 @@ export class Element {
   renderText(ctx: RenderContext, xPos: number, yPos: number): void {
     ctx.save();
     ctx.setFont(this.textFont);
-    ctx.fillText(this.text, xPos + this.xShift, yPos + this.yShift);
+    ctx.fillText(this.text, xPos + this.x + this.xShift, yPos + this.y + this.yShift);
+    this.children.forEach((child) => {
```
Comment (ronyeh) [inline src/element.ts:548]: Can we also clean up the grammar/formatting for the comment on lines 78-79 of this file? (e.g., "they are responsible for").

```js
  // all Element objects keep a list of children that they are responsible and which
  // inherit the style of their parents.
  protected children: Element[] = [];
```

Maybe something like:

```js
  // Element objects keep a list of children that they are responsible for.
  // Children inherit the style from their parents (see: setGroupStyle(s)).
  protected children: Element[] = [];
```

**Review by ronyeh — CHANGES_REQUESTED** [review]

File: src/clefnote.ts [hunk 2]
```diff
@@ -45,6 +45,6 @@ export class ClefNote extends Note {
 
     this.setRendered();
 
-    this.clef.renderText(ctx, this.getAbsoluteX(), stave.getYForLine(this.clef.line));
+    this.clef.renderText(ctx, this.getAbsoluteX() - this.x, stave.getYForLine(this.clef.line));
```
[the remark this answers is not in the document]
Comment (rvilarl) [reply src/clefnote.ts:48]: The idea is to generalize x and y in Element. I went for that when I saw noteheads (now inluded in this PR). getAbsoluteX inludes the x of the clef which is already considered in Element, I believe that Elements should also handle margins in the future.

File: src/element.ts (hunk 1 above)
Comment (rvilarl) [reply src/element.ts:548]: done

File: src/element.ts [hunk 3]
```diff
@@ -521,7 +544,11 @@ export class Element {
...
+      ctx.setFont(child.textFont);
+      ctx.fillText(child.text, xPos + child.x + child.xShift, yPos + child.y + child.yShift);
```
[the remark this answers is not in the document]
Comment (rvilarl) [reply src/element.ts:550]: No, they are independent. Same approach as Noteheads. They share the xPos and yPos given as parameter to RenderText.

File: src/element.ts [hunk 4]
```diff
@@ -75,8 +75,8 @@ export class Element {
     return Category.Element;
   }
 
-  // all Element objects keep a list of children that they are responsible and which
-  // inherit the style of their parents.
+  // Element objects keep a list of children that they are responsible for.
+  // Children inherit the style from their parents (see: setGroupStyle(s)).protected children: Element[] = [];
```
Comment (ronyeh) [inline src/element.ts:79]: There's a small copy/paste error here.

Comment (ronyeh) [reply src/element.ts:79]: Remvoe the "protected children..." at the end.

Comment (rvilarl) [reply src/element.ts]: Done.

Comment (rvilarl) [reply src/element.ts]: Done.

**Comment by ronyeh** [comment]
Thanks!

**Pull request merged**